core/src/main/java/io/fabric8/maven/core/util/ClassUtil.java

```diff
@@ -140,4 +167,27 @@ private static String convertToClass(String name, String prefix) {
         ret = ret.substring(0, name.length() - ".class".length());
         return ret.substring(prefix.length());
     }
+
+
+    public static URLClassLoader createProjectClassLoader(final MavenProject project, Logger log) {
+
+        try {
+
+            List<URL> compileJars = new ArrayList<>();
+
+            for (String element : project.getCompileClasspathElements()) {
+                compileJars.add(new File(element).toURI().toURL());
+            }
+
+            return new URLClassLoader(compileJars.toArray(new URL[compileJars.size()]),
+                    PluginServiceFactory.class.getClassLoader());
+
+        } catch (Exception e) {
```

Comment:
What kind of exception could occur here ? Should we really ignore it (i.e. only warn about) or shouldn't we break the build here because its supposed to be an internal error.